### DISCUSSION TITLE
Is anything being done to reverse updating everything???

### DISCUSSION THREAD

**Discussion by androticus** [Homebrew/discussions#4160] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 3.6.20
ORIGIN: https://github.com/Homebrew/brew
HEAD: 344d32bf7f40183359592ef81ca6486e0587981b
Last commit: 13 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: f355ac77b051dfe672d15016e627a5318c7216f6
Core tap last commit: 46 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.BZs3IvwlPl/org.macosforge.xquartz:0
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.8 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8_1/bin/ruby
CPU: octa-core 64-bit kabylake
Clang: 14.0.0 build 1400
Git: 2.37.1 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.86.0 => /usr/bin/curl
macOS: 13.2-x86_64
CLT: 11.3.0.32.62
Xcode: 14.2
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

[followed by an infinite number of console lines... none of this is relevant to the discussion]
```

### Description of issue

I was SHOCKED when I installed postgressql@15 and after doing that it UPDATED EVERYTHING ELSE I HAD INSTALLED! I'm not talking about dependencies to postgressql -- it updated EVERYTHING! node, spark, all kinds of other things, including cruft I should probably just remove, not bloat my system more with upgraded versions

WHO in their right mind thought this was a good idea??? Have the inmates completely taken over the asylum????

**Comment by gromgit** [comment]
https://docs.brew.sh/FAQ#why-does-brew-upgrade-formula-or-brew-install-formula-also-upgrade-a-bunch-of-other-stuff

**Comment by carlocab** [comment]
`brew install postgresql@15` will upgrade everything connected to `postgresql@15` via the dependency graph. In the case of `node`, both it and `postgresql@15` depend on `openssl@1.1`. Installing `postgresql@15` likely triggered an upgrade of `openssl@1.1` (because `postgresql@15` was built with a recent version of OpenSSL), which then triggers an upgrades of dependents of `openssl@1.1` (because what you have installed was built with older versions of OpenSSL).

> WHO in their right mind thought this was a good idea???

Someone who's thought about package management far longer and harder than you have. However, if you still think you know better, and are willing to put up with the risk of `brew install foo` or `brew upgrade bar`  breaking things, you can set `HOMEBREW_NO_INSTALLED_DEPENDENTS_CHECK`.
